diff --git a/image.tsx b/image.tsx
index 967b79b28959cf2f365738e462f4ce46a563ebae..a4a0efc5dbbf9d8c0d341d34b3a891f7ccdbca9e 100644
--- a/image.tsx
+++ b/image.tsx
@@ -1,4 +1,4 @@
-import React, { useState, type CSSProperties, type SyntheticEvent, type ImgHTMLAttributes } from 'react';
+import React, { useState, useEffect, type CSSProperties, type SyntheticEvent, type ImgHTMLAttributes, useRef } from 'react';
 import classNames from 'classnames';
 import styles from './image.module.scss';
 
@@ -67,9 +67,17 @@ export function Image({
   onError,
   ...rest
 }: ImageProps) {
+  const imgRef = useRef(null);
   const [isLoaded, setIsLoaded] = useState(typeof window === 'undefined');
   const [hasError, setHasError] = useState(false);
 
+  useEffect(() => {
+    // Check if image was already loaded (from cache)
+    if (imgRef.current?.complete) {
+      setIsLoaded(true);
+    }
+  }, []);
+
   const handleLoad = (event: SyntheticEvent<HTMLImageElement, Event>) => {
     setIsLoaded(true);
     if (onLoad) onLoad(event);
@@ -99,6 +107,7 @@ export function Image({
       data-error={hasError}
     >
       <img
+        ref={imgRef}
         className={classNames(styles.image, { [styles.imageLoaded]: isLoaded })}
         src={src}
         alt={alt}
